### ISSUE TITLE
Fix gateway-cli error handling and auto-completion

### ISSUE THREAD

**Issue by mayrf** [fedimint/fedimint#3966]
While working on #3880 I've noticed that errors returned from the handler functions in the gateway-cli are not processed properly, resulting in a generic error message ("Error: Bad status returned 500 Internal Server Error"). These should be handled similar to how fedimint-cli does it. Also, the auto-completion for gateway-cli subcommands does not seem to work.

**Comment by justinmoon** [comment]
Error-handling is a little more of a server-side thing. That gateway API usually doesn't provide useful error messages.

**Comment by mayrf** [comment]
Yeah, you're right, I thought that the server is actually returning useful errors messages and the gateway-cli is just not passing them through. Still, once we get the server to the return useful errors messages, the gateway should also pass them on and not just return the status code.

**Comment by elsirion** [comment]
TODO: test auto completion. Error reporting is a dup of #1999.
